### ISSUE TITLE
Intersection Observer API incorrectly detected as available

### ISSUE THREAD

**Issue by kaldonir** [verlok/lazyload#362]
On my Samsung TV, which is a 2018 model, the Intersection Observer API is not supported according to [Samsung's Web Engine Specifications](https://developer.samsung.com/tv/develop/specifications/web-engine-specifications). 

LazyLoad detects this API as available though, which causes images tagged as `lazy` to never be loaded at all. 

LazyLoad currently detects this via a check for `"IntersectionObserver" in window"`, which returns `true` on my TV. 

According to [this discussion in the IntersectionObserver repository](https://github.com/w3c/IntersectionObserver/issues/211), there are a few more indicators that might need to be checked. In particular, I had to do the following change to LazyLoad's detection for images to load as expected:

```javascript
   var supportsIntersectionObserver = runningOnBrowser && "IntersectionObserver" in window 
                                                       && "IntersectionObserverEntry" in window 
                                                       && "intersectionRatio" in window.IntersectionObserverEntry.prototype;
                                                       && "isIntersecting" in window.IntersectionObserverEntry.prototype;
```

I am not sure whether that's the best way to check whether the API is available, but the current one doesn't seem to be sufficient either, at least not on my TV.

**Comment by mrbusche** [comment]
FWIW, in Addy Osmani's blog post he says we should use [`'loading' in HTMLImageElement.prototype)`](https://addyosmani.com/blog/lazy-loading/)
Modernizr is using the same [`'loading' in HTMLImageElement.prototype`](https://github.com/Modernizr/Modernizr/blob/master/feature-detects/img/lazyloading.js#L18)

I'd be happy to make a PR with this change if that's agreed upon. Do we need more discussion @verlok ?

**Comment by verlok** [comment]
@kaldonir in addition, I think that the discussion you linked doesn't come to that conclusion. This was a detection method to get around a MS Edge version 15 bug. Now that version of MS Edge has been superseded and that issue is not around anymore. 

So my conclusion is still the same, please check if that full check solves the issue on your TV.

We had better use this other repository, to work on this problem.
https://github.com/verlok/IntersectionObserverDetect/

It deploys automatically and you can test it on your TV opening this link: https://www.andreaverlicchi.eu/IntersectionObserverDetect/

**Comment by verlok** [comment]
I guess that opening that link on your TV it says "yes", but it should say "no".

Then you could open a pull request to [this file](https://github.com/verlok/IntersectionObserverDetect/blob/master/index.html) using the full check you are proposing, and we could see if that works on your TV.

**Comment by verlok** [comment]
Actually I was curious and I made the change myself :D 
Could you check the URL on your TV now?

**Comment by kaldonir** [comment]
It only works if I keep the `"isIntersecting" in window.IntersectionObserverEntry.prototype" check. As soon as I remove that check, IntersectionObserver is detected as availabe on my TV.

**Comment by verlok** [comment]
That’s pretty much a bug as far as I know. Anyway, that’s cool, I can integrate the change I made in the other repo.

**Comment by verlok** [comment]
@kaldonir can you confirm that this https://www.andreaverlicchi.eu/IntersectionObserverDetect/ shows a NO on your TV?

**Comment by kaldonir** [comment]
@verlok Correct, it shows a NO.

**Comment by verlok** [comment]
Fixed in 12.0.2
